### DISCUSSION TITLE
Does supporting the Explicit Resource Management proposal hurt the ecosystem?

### DISCUSSION THREAD

**Discussion by JadenSimon** [nodejs/help#4645] [Q&A]
To preface: I'm not the type of person to create sensationalist discussions. My main concern here is the impact on the language and the developer experience, and I believe this needs more visibility.

By supporting this proposal, NodeJS is helping to entrench a feature that the author himself acknowledges is less valuable, compared to an alternative, in its current state.

From the [proposal author](https://github.com/tc39/proposal-explicit-resource-management/issues/244#issuecomment-2395175413):

> IMO, `defer` is only valuable over `using` in the short-term until the ecosystem catches up with `Symbol.dispose`/`Symbol.asyncDispose`, and that's already happening. NodeJS is already shipping `Symbol.asyncDispose` on a number of API objects.

This raises doubts about this feature in my mind. If anyone has their own experiences regarding `using`, I would really like to hear it. This is something the community should weigh in on. I personally found it to be clunky and confusing, especially with closures. I was very surprised when I found out `using` basically breaks with closures.

**Comment by cjihrig** [comment]
The proposal is already at Stage 3 at TC39 and is available in Chrome, Deno, and probably more environments that I did not check.

FYI - Node recently [removed its polyfills](https://github.com/nodejs/node/pull/55276).

**Reply by JadenSimon** [reply]
Yeah it's been in stage 3 for over a year now I think. I hadn't given it a serious look until last week. 

> FYI - Node recently https://github.com/nodejs/node/pull/55276.

Interesting, so v8 has the symbols?

**Comment by abhi-sojitra** [comment]
I appreciate you bringing this up! It’s interesting to hear that using might not be as beneficial as Symbol.dispose/Symbol.asyncDispose. If the community is leaning toward asyncDispose, it seems like a smarter long-term choice.

I’ve had my own struggles with using, especially with closures—it can get pretty messy and confusing. I’d love to hear more from others about their experiences, too!
